### PR TITLE
Normalization

### PR DESCRIPTION
**Pull request by sfraczek**
and cleanup.

This is the diff to see all changes vs initial place of branching
https://github.com/sfraczek/models/compare/develop...sfraczek:normalization

### CODE REVIEW

File: PaddleCV/image_classification/run.sh [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+# python eval.py \
+#        --model=ResNet50 \
+#        --batch_size=16 \
+#        --class_dim=1000 \
+#        --image_shape=3,224,224 \
+#        --with_mem_opt=True \
+#        --use_gpu=True \
+#        --pretrained_model=/mnt/drive/recovered/ResNet50_pretrained
+
+## AEON EVAL (LOOPING)
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/run.sh]: I'd split to separate file entries corresponding to Aeon.

File: PaddleCV/image_classification/train2.py [hunk 2]
```diff
@@ -405,6 +398,9 @@ def if_exist(var):
                         .format(pass_id, test_batch_id, "%.5f"%loss,"%.5f"%acc1, "%.5f"%acc5,
                                 "%2.2f sec" % period))
                 sys.stdout.flush()
+            test_batch_id += 1
+        except fluid.core.EOFException:
+            test_py_reader.reset()
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train2.py]: Are you sure it is correct? Where the `test_py_reader` is defined? Why after all you've been making changes in this file if we're not using Aeon here?

File: PaddleCV/image_classification/train_aeon.py [hunk 3]
```diff
@@ -38,19 +38,19 @@
...
 add_arg('momentum_rate',    float, 0.9,                  "momentum_rate.")
-add_arg('use_label_smoothing',      bool,      False,        "Whether to use label_smoothing or not")
-add_arg('label_smoothing_epsilon',      float,     0.2,      "Set the label_smoothing_epsilon parameter")
 add_arg('lower_scale',      float,     0.08,      "Set the lower_scale in ramdom_crop")
 add_arg('lower_ratio',      float,     3./4.,      "Set the lower_ratio in ramdom_crop")
 add_arg('upper_ratio',      float,     4./3.,      "Set the upper_ratio in ramdom_crop")
 add_arg('resize_short_size',      int,     256,      "Set the resize_short_size")
-add_arg('is_distill',       bool,  False,        "is distill or not")
-
+add_arg('cache_dir',           str, "",              "Place where aeon will store cache")
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train_aeon.py]: Please add some default value.

File: PaddleCV/image_classification/reader_aeon.py [hunk 4]
```diff
@@ -1,42 +1,44 @@
...
         "bgr_to_rgb": True
     }
 
     label_config = {"type": "label", "binary": False}
 
+    config = dict()
+    config['random_seed'] = random_seed
+    config['decode_thread_count'] = thread_count
+    config['manifest_root'] = data_dir
+    config['cache_directory'] = cache_dir
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/reader_aeon.py]: We may check whether `cache_dir` exists and if not create such directory.

File: PaddleCV/image_classification/train_aeon.py [hunk 5]
```diff
@@ -38,19 +38,19 @@
...
 add_arg('lr_strategy',      str,   "piecewise_decay",    "Set the learning rate decay strategy.")
 add_arg('model',            str,   "SE_ResNeXt50_32x4d", "Set the network to use.")
+add_arg('enable_ce',        bool,  False,                "If set True, enable continuous evaluation job.")
 add_arg('data_dir',         str,   "./data/ILSVRC2012/",  "The ImageNet dataset root dir.")
 add_arg('fp16',             bool,  False,                "Enable half precision training with fp16." )
 add_arg('scale_loss',       float, 1.0,                  "Scale loss for fp16." )
 add_arg('l2_decay',         float, 1e-4,                 "L2_decay parameter.")
 add_arg('momentum_rate',    float, 0.9,                  "momentum_rate.")
-add_arg('use_label_smoothing',      bool,      False,        "Whether to use label_smoothing or not")
-add_arg('label_smoothing_epsilon',      float,     0.2,      "Set the label_smoothing_epsilon parameter")
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train_aeon.py]: I think we should get this back. We haven't used it, however I'm not sure whether this is not used at all by anyone from Baidu.

File: PaddleCV/image_classification/train_aeon.py [hunk 6]
```diff
@@ -38,19 +38,19 @@
...
 add_arg('lower_scale',      float,     0.08,      "Set the lower_scale in ramdom_crop")
 add_arg('lower_ratio',      float,     3./4.,      "Set the lower_ratio in ramdom_crop")
 add_arg('upper_ratio',      float,     4./3.,      "Set the upper_ratio in ramdom_crop")
 add_arg('resize_short_size',      int,     256,      "Set the resize_short_size")
-add_arg('is_distill',       bool,  False,        "is distill or not")
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train_aeon.py]: As above, I think we should leave it here.

File: PaddleCV/image_classification/train_aeon.py [hunk 7]
```diff
@@ -160,25 +160,21 @@ def optimizer_setting(params):
 
     return optimizer
 
-
-def calc_loss(epsilon,label,class_dim,softmax_out,use_label_smoothing):
-    if use_label_smoothing:
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train_aeon.py]: I think we should do as small changes as we can during integration. We should leave this in script.

File: PaddleCV/image_classification/train_aeon.py [hunk 8]
```diff
@@ -194,18 +190,12 @@ def net_config(image, model, args, is_train, label=0, y_a=0, y_b=0, lam=0.0):
         acc_top5 = fluid.layers.accuracy(input=out0, label=label, k=5)
 
     else:
-        if not args.is_distill:
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train_aeon.py]: I think we should do as small changes as we can during integration. We should leave this in script.

File: PaddleCV/image_classification/train_aeon2.py [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+            else:
+                loss, acc1, acc5, lr = train_exe.run(fetch_list=train_fetch_list, feed=feeder.feed(data))
+
+
+            acc1 = np.mean(np.array(acc1))
+            acc5 = np.mean(np.array(acc5))
+            train_info[1].append(acc1)
+            train_info[2].append(acc5)
+
+            t2 = time.time()
```
Comment (arogowie-intel) [inline PaddleCV/image_classification/train_aeon2.py]: I would move this line before calculating accuracy mean.

File: PaddleCV/image_classification/eval.py [hunk 10]
```diff
@@ -113,15 +105,13 @@ def eval(args):
...
-            break
 
     test_loss = np.sum(test_info[0]) / cnt
     test_acc1 = np.sum(test_info[1]) / cnt
     test_acc5 = np.sum(test_info[2]) / cnt
 
     print("Test_loss {0}, test_acc1 {1}, test_acc5 {2}".format(
-        "%.5f" % test_loss, "%.5f" % test_acc1, "%.5f" % test_acc5))
+        "%.5f"%test_loss, "%.5f"%test_acc1, "%.5f"%test_acc5))
     sys.stdout.flush()
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/eval.py]: I didn't change that because the actual changes are reflected here:
https://github.com/sfraczek/models/compare/develop...sfraczek:normalization

This PR is branched from aeon-integration

[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/eval.py]: I'm just removing the change that I did previously

File: PaddleCV/image_classification/eval_aeon.py [hunk 11]
```diff
@@ -26,10 +26,12 @@
 add_arg('pretrained_model', str,  None,                "Whether to use pretrained model.")
 add_arg('model',            str,  "SE_ResNeXt50_32x4d", "Set the network to use.")
 add_arg('resize_short_size', int, 256,                "Set resize short size")
-add_arg('iterations',       int, 50000,                  "Quit after this many iterations")
+add_arg('data_dir',            str, "./data/ILSVRC2012/", "The ImageNet dataset root dir.")
+add_arg('cache_dir',           str, "",                   "Place where aeon will store cache")
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/eval_aeon.py]: "" is the default here which means that cache is not used.

File: PaddleCV/image_classification/reader_aeon.py [hunk 12]
```diff
@@ -93,20 +77,58 @@ def val_reader(settings):
...
     dl = DataLoader(config)
     return dl
+
+
+def train(settings, batch_size, drop_last=False):
+    # Batch size check
+    batch_size = int(batch_size)
+    if batch_size <= 0:
+        raise ValueError(
+            "batch_size should be a positive integeral value, but got batch_size={}"
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/reader_aeon.py]: it still should be positive. Should I de;ete word `integral`?

File: PaddleCV/image_classification/reader_aeon.py [hunk 13]
```diff
@@ -93,20 +77,58 @@ def val_reader(settings):
...
+            .format(batch_size))
+    reader = train_reader(settings, batch_size)
+
+    def func():
+        for tup in reader:
+            #  batch = {k: v for k, v in tup}
+            #  images = batch['image']
+            #  labels = batch['label']
+            #  yield zip(images, labels)
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/reader_aeon.py]: ok

File: PaddleCV/image_classification/reader_aeon.py [hunk 14]
```diff
@@ -93,20 +77,58 @@ def val_reader(settings):
...
+            if len(tup[0][1]) == batch_size:
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/reader_aeon.py]: ok

File: PaddleCV/image_classification/train_aeon.py (hunk 5 above)
Comment (sfraczek) [reply PaddleCV/image_classification/train_aeon.py]: ok

File: PaddleCV/image_classification/train_aeon.py (hunk 8 above)
Comment (sfraczek) [reply PaddleCV/image_classification/train_aeon.py]: i think is_distill wouldn't work

File: PaddleCV/image_classification/train_aeon.py (hunk 7 above)
Comment (sfraczek) [reply PaddleCV/image_classification/train_aeon.py]: ok

File: PaddleCV/image_classification/train_aeon.py (hunk 6 above)
Comment (sfraczek) [reply PaddleCV/image_classification/train_aeon.py]: i think it wouldnt work

File: PaddleCV/image_classification/train_aeon2.py (hunk 9 above)
Comment (sfraczek) [reply PaddleCV/image_classification/train_aeon2.py]: ok

File: PaddleCV/image_classification/reader_aeon.py [hunk 15]
```diff
@@ -93,20 +77,58 @@ def val_reader(settings):
...
+            .format(batch_size))
+    reader = val_reader(settings, batch_size)
+
+    def func():
+        for tup in reader:
+            #  batch = {k: v for k, v in tup}
+            #  images = batch['image']
+            #  labels = batch['label']
+            #  yield zip(images, labels)
+            if len(tup[0][1]) == batch_size:
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/reader_aeon.py]: Can you explain how or help with that?

File: PaddleCV/image_classification/reader_aeon.py [hunk 16]
```diff
@@ -93,20 +77,58 @@ def val_reader(settings):
...
+                yield zip(tup[0][1], tup[1][1])
```
[the remark this answers is not in the document]
Comment (sfraczek) [reply PaddleCV/image_classification/reader_aeon.py]: added explanation
